cmd/entrypoint/main.go

```diff
@@ -75,14 +77,16 @@ type RealWaiter struct{}
 
 var _ entrypoint.Waiter = (*RealWaiter)(nil)
 
-func (*RealWaiter) Wait(file string) error {
+func (*RealWaiter) Wait(file string, expectContent bool) error {
 	if file == "" {
 		return nil
 	}
 	for ; ; time.Sleep(time.Second) {
 		// Watch for the post file
-		if _, err := os.Stat(file); err == nil {
-			return nil
+		if info, err := os.Stat(file); err == nil {
+			if !expectContent || info.Size() > 0 {
+				return nil
+			}
 		} else if !os.IsNotExist(err) {
 			return xerrors.Errorf("Waiting for %q: %w", file, err)
 		}
```

Comment:
I'd love to see some/any unit tests for this if possible :innocent: 

(i think what i'd do is take the bulk of the `Wait` function, put it in a separate function - probably in  pkg/entrypoint/entrypointer.go - and test that)

Reply:
I've added unit tests for the `RealWaiter` type.  I wasn't entirely sure how best to break this out into the `entrypoint` package so I've just added them initially alongside `main.go` in a `main_test.go`. Happy to move them elsewhere but might need a bit more guidance on how best to separate things out.